hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/OzoneAddress.java

```diff
@@ -195,20 +197,19 @@ public OzoneClient createClientForS3Commands(
         throw new OzoneClientException("Service ID specified does not match" +
             " with " + OZONE_OM_SERVICE_IDS_KEY + " defined in the " +
             "configuration. Configured " + OZONE_OM_SERVICE_IDS_KEY + " are" +
-            conf.getTrimmedStringCollection(OZONE_OM_SERVICE_IDS_KEY));
+            serviceIds);
       }
-    } else {
+    } else if (serviceIds.size() > 1) {
       // If om service id is not specified, consider it as a non-HA cluster.
       // But before that check if serviceId is defined. If it is defined
-      // throw an error om service ID needs to be specified.
-      if (OmUtils.isServiceIdsDefined(conf)) {
-        throw new OzoneClientException("Service ID must not"
-            + " be omitted when " + OZONE_OM_SERVICE_IDS_KEY + " is defined. " +
-            "Configured " + OZONE_OM_SERVICE_IDS_KEY + " are " +
-            conf.getTrimmedStringCollection(OZONE_OM_SERVICE_IDS_KEY));
+      // and has length = 1, then take that ID as default.
+      // otherwise throw an error "om service ID needs to be specified."
```

Comment:
Nit: I think this comment no longer describes the code, can be removed.

Reply:
I have modified the comment to describe the code now